packages/handlersjs-core/lib/storage/memory-store.spec.ts

```diff
@@ -18,111 +18,123 @@ describe('MemoryStore', () => {
 
   });
 
-  it('should not iterate over keys pairs when it is empty', async () => {
+  describe('constructor', () => {
 
-    const elements = [];
-    for await(const entry of memoryStore.entries()) elements.push(entry);
+    it('can be initialized with values', async () => {
 
-    expect(elements.length).toBe(0);
+      const initialData: [keyof TestInterface, TestInterface[keyof TestInterface]][] = [
+        [ 'key1', 4 ],
+        [ 'key2', '123' ],
+        [ 'key4', [ '', '', '321' ] ],
+      ];
 
-  });
+      memoryStore = new MemoryStore(initialData);
 
-  it('should not have keys that were not added', async () => {
+      expect(await memoryStore.get('key1')).toEqual(4);
+      expect(await memoryStore.get('key2')).toEqual('123');
+      expect(await memoryStore.get('key4')).toEqual([ '', '', '321' ]);
 
-    expect(await memoryStore.has('key1')).toBeFalsy();
-    expect(await memoryStore.get('key1')).toBeUndefined();
-    expect(await memoryStore.latestUpdate('key1')).toBeUndefined();
-    expect(await memoryStore.hasUpdate('key1', Date.now())).toBeUndefined();
+    });
 
   });
 
-  it('should get a value that has been set', async () => {
+  describe('set(), has() & get()', () => {
```

Comment:
I know it is shorter this way, but it is more easy to overview when each of these tests is in the descbribe block of its own function.